[WIP][MIG][14.0] oauth_provider

remytms opened the pull request:
This is a migration of the module `oauth_provider` that comes from the 9.0 version (repo server-tools) and based on the PR for this module in 12.0 (#405 ).

Work is in progress for now.

Commit:
[ADD] Add oauth_provider module

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

[UPD] Update oauth_provider.pot

Commit:
[FIX] oauth_provider: Fix tests when using last oauthlib version

OCA-git-bot committed:
[ADD] icon.png

ccomb committed:
typo

hbrunn committed:
[MIG] oauth_provider: Migration to 12.0

hbrunn committed:
[ADD] auth method oauth_provider

remytms committed:
[MIG] oauth_provider: @api.multi

remytms committed:
[FIX] oauth_provider: dependencies of a compute method

remytms committed:
[MIG] oauth_provider: change usage of sudo

remytms committed:
[MIG] oauth_provider: view_type

remytms committed:
[MIG] oauth_provider: black, prettier, isort

remytms committed:
[MIG] oauth_provider: split readme

remytms committed:
[MIG] oauth_provider: add requirements and setup

remytms committed:
[MIG] oauth_provider: update manifest

hbrunn commented:
please make this a draft PR until it's ready for review. And better do the black, prettier, isort really as first commit, then it's simpler to squash your other commits

remytms committed:
[MIG] oauth_provider: make module installable

remytms committed:
[MIG] oauth_provider: change datetime call

remytms committed:
[FIX] oauth_provider: error using module with safe_eval

remytms committed:
[FIX] oauth_provider: validator

Lines removed tries to write on `client_id` attribute of the
`oauth.provider.client` which do not have such an attribute.
Theses sentences seams useless, so I removed it.

remytms committed:
[FIX] oauth_provider: dependencies

This module uses the method `get_frontend_session_info()` from
`ir.http`, and this method is defined in the web module.

remytms commented:
To fix this module is 14.0 get inspired by the [PR in 16.0](https://github.com/OCA/server-auth/pull/795).

This PR introduce a bug that is fixed [here](https://github.com/OCA/server-auth/pull/795/commits/978f0e06f7524d552c03e8d3a492d2eb27e5ee16).
